update_image.pl

```diff
@@ -443,29 +455,37 @@ sub processConvergedSections {
     # Populate OCMBFW partition if it exists in the layout
     if(checkForPnorPartition("OCMBFW", $parsed_pnor_layout))
     {
-        if(-e $ocmbfw_original_filename)
-        {
-            #Add header to delivered image
-            my $date = `date`;
-            run_command("$hb_image_dir/pkgOcmbFw.pl --unpackagedBin $ocmbfw_original_filename --packagedBin $ocmbfw_original_filename.header --timestamp \"$date\" --vendorVersion \"$ocmbfw_version\" --vendorUrl \"$ocmbfw_url\"");
-
-            #Verify ocmbfw header
-            run_command("$hb_image_dir/pkgOcmbFw.pl --verify --packagedBin $ocmbfw_original_filename.header");
-            $sections{OCMBFW}{in}    = "$ocmbfw_original_filename.header";
-        }
-        else
+        if(!(-e $ocmbfw_original_filename))
         {
             print "WARNING: OCMBFW binary not found, generating blank binary (w/ valid header) instead\n";
             #Create blank 4k image
             run_command("dd if=/dev/zero of=$ocmbfw_original_filename bs=1024 count=4");
-            #Add header to blank image
-            my $date = `date`;
-            run_command("$hb_image_dir/pkgOcmbFw.pl --unpackagedBin $ocmbfw_original_filename --packagedBin $ocmbfw_original_filename.header --timestamp \"$date\" --vendorVersion \"$ocmbfw_version\" --vendorUrl \"$ocmbfw_url\"");
+        }
 
-            # verify ocmbfw header
-            run_command("$hb_image_dir/pkgOcmbFw.pl --verify --packagedBin $ocmbfw_original_filename.header");
-            $sections{OCMBFW}{in}    = "$ocmbfw_original_filename.header";
+        # If the Odyssey pak files exist, package those up into the
+        # image with the odyssey OCMBFW PNOR partition layout. If not,
+        # use a layout that only packages the Explorer firmware.
+
+        my $json_layout_template = "$hb_image_dir/ocmbfw-layout-exp-only.json.template";
+
+        if (-e $ody_bldr_pak_file)
+        {
+            $json_layout_template = "$hb_image_dir/ocmbfw-layout.json.template";
         }
+
+        # Preprocess the JSON file to replace variables.
+
+        run_command("cpp \"-DUNPKGD_EXP_FW_IMG=\\\"$ocmbfw_original_filename\\\"\" \\
+                         \"-DUNPKGD_ODY_BLDR_IMG=\\\"$ody_bldr_pak_file\\\"\" \\
```

Comment:
Will we get an error here if ody_bldr_pak_file DNE?

Reply:
yes, but that's ok because the JSON file (selected on line 473) just won't contain a specification that uses that file, so that will never be accessed.

Reply:
The compiler is upset about it though:

```cpp: warning: ‘-x c’ after last input file has no effect
cpp: fatal error: no input files
compilation terminated.```

https://hostboot-jenkins.swg-devops.com/job/Hostboot/job/OpenPower/job/GHPR/job/build-p10-pnor-public-github/20/CONFIG=p10ebmc,OS=rhel7/console

Reply:
yeah, it's because the files that it will process aren't provided by the version of hostboot that this op-build is using. We need a couple of hostboot commits to be pulled in before this will pass CI/it's mergeable.